changelogs/unreleased/gh-7486-deprecate-replication_synchro_timeout.md

```diff
@@ -0,0 +1,15 @@
+## feature/replication
+
+* This patch allows the `replication_synchro_timeout` option to be set to 0,
+  which means an infinite timeout.
+* A new compat option `box_cfg_replication_synchro_timeout` has been added.
+  This option is `'old'` by default. `box_cfg_replication_synchro_timeout` can
+  only be set to `'new'` if `replication_synchro_timeout` is already set to 0.
+  With new behavior, option `replication_synchro_timeout` deprecated (it can
```

Comment:
```suggestion
  With the new behavior, option `replication_synchro_timeout` is deprecated (it can
```

Reply:
@Astronomax please address these comments in the scope of #9869.

Reply:
Fixed.